### PR TITLE
DataGrid - Should hide filter row menu after losing it's focus

### PR DESCRIPTION
**Pull request by igormalzew**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.filter_row.js [hunk 1]
```diff
@@ -538,11 +543,15 @@ var ColumnHeadersViewFilterRowExtender = (function() {
...
-
                     eventsEngine.trigger($menu, "blur");
-                    menu && menu.option("focusedElement", null);
-                    isCellWasFocused && that._focusEditor($editorContainer);
+                    restoreFocus();
+                },
+                onContentReady: function(e) {
+                    eventsEngine.on($menu, "blur", () => {
+                        var menu = e.component;
+                        menu._hideSubmenu(menu._visibleSubmenu);
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.filter_row.js:552]: Is it posiible not use private API of Menu widget?

Comment (igormalzew) [reply js/ui/grid_core/ui.grid_core.filter_row.js:552]: There are no public methods to close the menu. I also tried to trigger various events on the menu element, but this did not produce results.

File: testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js [hunk 2]
```diff
@@ -178,6 +178,33 @@ QUnit.test("Accessibility columns id should not set for columns editors (T710132
...
+    $menu.focus();
+
+    menuInstance = $menu.dxMenu("instance");
+    $root = $(menuInstance.itemElements().get(0));
+    menuInstance._showSubmenu($root);
+
+    $menu.trigger("blur");
+
+    // assert
+    assert.notOk(!!menuInstance._visibleSubmenu, "submenu is hidden");
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js]: It is bad assert. If _visibleSubmenu is renamed test will not fail. It is better to check menu overlay element visiibility

**Review by vconst — CHANGES_REQUESTED** [review]

File: testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js (hunk 2 above)
Comment (igormalzew) [reply testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js]: I added double check for the submenu visibility which covers the "private field is renamed" case.

**Review by vconst — APPROVED** [review]

**Review by AntonSermyazhko — APPROVED** [review]

**Pull request merged**